### PR TITLE
feat: adding ability for for charts to be pulled without HTTPS

### PR DESCRIPTION
**Pull request by hoangelos**
accomplished by:

    - Adding PlainHttp attribute to RepositorySpec., HelmDefault, ReleaseSpec
    - Adding UnitTests for getOCIChart Flags.
    - Adding funciton and unitTests for getChartDownload
    - Changing and refactoring how Flags are added to getOCIChart.

    Resolves #1224

### CODE REVIEW

**Comment by yxxhero** [comment]
@hoangelos please fix ci issues.

**Comment by hoangelos** [comment]
@yxxhero  not sure why this test is failing.  Any help?

**Comment by yxxhero** [comment]
DCO issue

**Comment by hoangelos** [comment]
@yxxhero  address DCO and a lint issue.  I rebased too.  Also, you can see  there's a failing unit test in code I didn't get near.  Some difference in the hash function here.

**Comment by yxxhero** [comment]
微信图片_20231228081618
please update the values.

File: pkg/state/state.go [hunk 1]
```diff
@@ -213,6 +215,7 @@ type RepositorySpec struct {
 	Keyring         string `yaml:"keyring,omitempty"`
 	PassCredentials bool   `yaml:"passCredentials,omitempty"`
 	SkipTLSVerify   bool   `yaml:"skipTLSVerify,omitempty"`
+	PlainHttp       bool   `yaml:"plainHttp,omitempty"`
```
Comment (yxxhero) [inline pkg/state/state.go:218]: I can't see any code that handle thie field. @hoangelos

Comment (yxxhero) [reply pkg/state/state.go:218]: @hoangelos

Comment (hoangelos) [reply pkg/state/state.go:218]: @yxxhero  I pinged below on where it is.

File: pkg/state/state.go [hunk 2]
```diff
@@ -2542,6 +2549,16 @@ func (st *HelmState) appendChartDownloadTLSFlags(flags []string, release *Releas
 	return flags
 }
 
+func (st *HelmState) appendChartDownloadPlainHttpFlags(flags []string, release *ReleaseSpec) []string {
+	switch {
+	case release.PlainHttp:
+		flags = append(flags, "--plain-http")
+	case st.HelmDefaults.PlainHttp:
```
[the remark this answers is not in the document]
Comment (yxxhero) [reply pkg/state/state.go:2548]: RepositorySpec.PlainHttp? @hoangelos

[the remark this answers is not in the document]
Comment (yxxhero) [reply pkg/state/state.go:2548]: @hoangelos

**Comment by yxxhero** [comment]
@hoangelos ping

**Comment by yxxhero** [comment]
any updates?

**Comment by yxxhero** [comment]
@hoangelos friendly ping.

**Comment by yxxhero** [comment]
see: https://github.com/helmfile/helmfile/pull/1672

**Pull request closed without merging**